### PR TITLE
ARTEMIS-711 Fix handling of 'bare' header

### PR DESCRIPTION
**Pull request by jbertram**

### CODE REVIEW

File: artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/proton/plug/ActiveMQProtonConnectionCallback.java [hunk 1]
```diff
@@ -138,4 +146,8 @@ public AMQPSessionCallback createSessionCallback(AMQPConnectionContext connectio
       return new ProtonSessionIntegrationCallback(this, manager, connection, this.connection, closeExecutor);
    }
 
+   @Override
+   public void sendSASLSupported() {
```
Comment (clebertsuconic) [inline artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/proton/plug/ActiveMQProtonConnectionCallback.java]: Is there a link you could add here as a Doc? just for future reference?

File: tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/openwire/util/Wait.java [hunk 2]
```diff
@@ -41,13 +41,13 @@ public static boolean waitFor(final Condition condition, final long duration) th
 
    public static boolean waitFor(final Condition condition,
                                  final long duration,
-                                 final int sleepMillis) throws Exception {
```
Comment (clebertsuconic) [inline tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/openwire/util/Wait.java]: This doesn't seem related?


It's a minor change and I don't mind.. just checking if I'm missing anything.

**Comment by clebertsuconic** [comment]
@jbertram checkstyles

File: tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/openwire/util/Wait.java (hunk 2 above)
Comment (jbertram) [inline tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/openwire/util/Wait.java]: It's used by org.apache.activemq.artemis.tests.integration.proton.ProtonTestForHeader#testSimpleBytes.
